x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/QueryBuilderResolver.java

```diff
@@ -98,9 +97,9 @@ private Set<FullTextFunction> fullTextFunctions(LogicalPlan plan) {
     }
 
     public Set<String> indexNames(LogicalPlan plan) {
-        Holder<Set<String>> indexNames = new Holder<>();
```

Comment:
I have little mileage on this part of the code, but the change here matches with the (subtle) [change](https://github.com/elastic/elasticsearch/pull/121260/files#diff-2a0fbe692bfc91f764a424e8035abce947534e2376cb7ade9c6ae55cbc5f2667R68) from https://github.com/elastic/elasticsearch/pull/121260 and looks correct to me.

Reply:
Yes, so because `indexNames` here did not include the name of the index that actually had the `semantic_text` field, the query builder rewriter does not rewrite the `match` query to the underlying `semantic` query.